lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -0,0 +1,148 @@
+    std::shared_ptr<SessionStateEnforcer> m5genforcer, SessionStore& session_store)
+    : session_store_(session_store),
+      m5g_enforcer_(m5genforcer) {}
+
+/*Building session config with required parameters*/  //TODO on SubscriberID how to assigne or manage
+SessionConfig SetMessageManagerHandler::m5g_build_session_config(
+              const SetSMSessionContext& request) {
+  SessionConfig cfg ;
+  /*copying pnly 5G specific data to respective elements*/
+  cfg.common_context         = request.common_context();
+  cfg.rat_specific_context   = request.rat_specific_context();
+
+  return cfg;
+}
+
+/* Handling set message from AMF
+ * check if it is INITIAL_REQUEST or EXISTING_PDU_SESSION
+ * if it is INITIAL_REQUEST need to create the session context in SessionMap
+ * and write to memory by SessionStore
+ * If EXISTING_PDU_SESSION, check on incoming session state and version
+ * accordingly take action.
+ * As per ASYNC method, response_callback is included but not functional
+ * */
+
+void SetMessageManagerHandler::SetAmfSessionContext(
+		ServerContext* context,
+		const SetSMSessionContext* request,
+		std::function<void(Status, SmContextVoid)> response_callback)
+{
+   auto& request_cpy = *request;
+   //Print the message from AMF
+   PrintGrpcMessage(static_cast<const google::protobuf::Message&>(request_cpy));
+   response_callback(Status::OK, SmContextVoid());
+   /*The Event Based main_thread invocation and runs to handle session state*/ 
+   m5g_enforcer_->get_event_base().runInEventBaseThread (
+      [this, response_callback, request_cpy]() {
+      //[this, context, response_callback, request_cpy]() {
+   std::cerr << __LINE__ << " Entered event loop therad " << __FUNCTION__ << " processing message \n";
+   //extract values from proto
+   auto  imsi = request_cpy.common_context().sid().id();
+   std::string dnn = request_cpy.common_context().apn(); //may not required for demo-1
+   std::string session_id = id_gen_.gen_session_id(imsi);
+
+   MLOG(MDEBUG) << "Requested imsi from UE: " << imsi
+                 << " Generated sessioncontext ID" << session_id;
+   std::cerr << __LINE__ << " Requested imsi from UE " << imsi << "\n";
+   /*reach complete message from proto message*/
+   SessionConfig cfg = m5g_build_session_config(request_cpy);
+   std::cerr << __LINE__ << " m5g_build_session_config called \n";
+
+   /*check if it's initial message*/
+   if((cfg.rat_specific_context.m5gsm_session_context().rquest_type() == INITIAL_REQUEST) &&
+      (cfg.common_context.sm_session_state() == CREATING_0)) {
+      /*it's new UE establisment request and need to create the session context*/
+      	 MLOG(MDEBUG) << "AMF request type INITIAL_REQUEST and session state CREATING";
+      	 std::cerr << __LINE__ << " AMF request type INITIAL_REQUEST and session state CREATING \n";
+         /* read the SessionMap from global session_store 
+	  * if not found it will add this imsi
+	  */
+	 //SessionRead req = {imsi};

```

Comment:
Please remove commented out lines of code

Reply:
Will adapt